src/python/zquantum/qaoa/ansatzes/warm_start_ansatz.py

```diff
@@ -0,0 +1,87 @@
+from zquantum.core.interfaces.ansatz import Ansatz, ansatz_property
+from zquantum.core.circuit import Circuit, Qubit, create_layer_of_gates
+from zquantum.core.evolution import time_evolution
+from zquantum.core.openfermion import qubitop_to_pyquilpauli, change_operator_type
+
+from openfermion import QubitOperator, IsingOperator
+from openfermion.utils import count_qubits
+from typing import Union, Optional
+import numpy as np
+import sympy
+from overrides import overrides
+
+
+class WarmStartQAOAAnsatz(Ansatz):
+
+    supports_parametrized_circuits = True
+    cost_hamiltonian = ansatz_property("cost_hamiltonian")
+    thetas = ansatz_property("thetas")
+
+    def __init__(
+        self,
+        number_of_layers: int,
+        cost_hamiltonian: Union[QubitOperator, IsingOperator],
+        thetas: np.ndarray,
+    ):
+        """TODO
+
+        Args:
+            number_of_layers: number of layers of the ansatz. Also refered to as "p" in the paper.
+            cost_hamiltonian: Hamiltonian representing the cost function
+            mixer_hamiltonian: Mixer hamiltonian for the QAOA. If not provided, will default to basic operator consisting of single X terms.
+
+        Attributes:
+            number_of_qubits: number of qubits required for the ansatz circuit.
+            number_of_params: number of the parameters that need to be set for the ansatz circuit.
+        """
+        super().__init__(number_of_layers)
+        self._cost_hamiltonian = cost_hamiltonian
+        self._thetas = thetas
+
+    @property
+    def number_of_qubits(self):
+        """Returns number of qubits used for the ansatz circuit."""
+        return count_qubits(change_operator_type(self._cost_hamiltonian, QubitOperator))
+
+    @property
+    def number_of_params(self) -> int:
+        """Returns number of parameters in the ansatz."""
+        return 2 * self.number_of_layers
+
+    @overrides
+    def _generate_circuit(self, params: Optional[np.ndarray] = None) -> Circuit:
+        """Returns a parametrizable circuit represention of the ansatz.
+        By convention the initial state is taken to be the |+..+> state and is
```

Comment:
This is different than the initial state prepared in the code which I believe uses parametrized Ry's on each qubit

Reply:
Fixed in https://github.com/zapatacomputing/z-quantum-qaoa/pull/31/commits/0bdf13c508c63ae5f8f26ed819fb7b57b4ba9110